src/channel.rs

```diff
@@ -76,14 +79,43 @@ impl Channel {
     pub fn read_from<R: ::std::io::BufRead>(reader: R) -> Result<Channel, Error> {
         let mut reader = XmlReader::from_reader(reader).trim_text(true);
         let mut in_rss = false;
+        let mut namespaces = HashMap::new();
 
         while let Some(e) = reader.next() {
             match e {
                 Ok(Event::Start(element)) => {
                     match element.name() {
-                        b"rss" if !in_rss => in_rss = true,
+                        b"rss" if !in_rss => {
+                            for attr in element.attributes().with_checks(false) {
+                                if let Ok(attr) = attr {
+                                    let split =
+                                        attr.0.splitn(2, |b| *b == b':').collect::<Vec<_>>();
+                                    if split.len() != 2 {
+                                        continue;
+                                    }
+
+                                    let ns = unsafe { split.get_unchecked(0) };
+                                    if ns != b"xmlns" {
+                                        continue;
+                                    }
+
+                                    let name = unsafe { split.get_unchecked(1) };
+                                    if name == b"itunes" || name == b"dc" {
+                                        continue;
+                                    }
+
+                                    let key = try!(str::from_utf8(name)).to_string();
```

Comment:
We're not able to use `String::from_utf8` here since `name` is a `&[u8]` and `String::from_utf8` requires a `Vec<u8>`.